### PR TITLE
Basic version of JMS endpoint generator (happy path)

### PR DESCRIPTION
**Pull request by jaceko**

### CODE REVIEW

File: adapters/adapters-test-utils/pom.xml [hunk 1]
```diff
@@ -29,4 +27,4 @@
             <artifactId>raml-parser</artifactId>
         </dependency>
     </dependencies>
-</project>
+</project>
```
Comment (purple52) [inline adapters/adapters-test-utils/pom.xml]: This file only has formatting changes - can it be excluded from the changeset?

File: adapters/jms-adapter-generator/pom.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            <artifactId>javaee-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>uk.gov.justice</groupId>
+            <artifactId>raml-generator-core</artifactId>
+            <version>1.0.0-SNAPSHOT</version>
```
Comment (purple52) [inline adapters/jms-adapter-generator/pom.xml]: Should use a property, not a hard coded version number.

File: adapters/jms-adapter-generator/pom.xml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.raml</groupId>
+            <artifactId>raml-parser</artifactId>
+            <version>0.8.12</version>
```
Comment (purple52) [inline adapters/jms-adapter-generator/pom.xml]: Remove version number - it is now in common-bom

File: adapters/jms-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/JmsEndpointGenerator.java [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+     * 
+     * @param raml
+     *            - the RAML object
+     * @param generatorConfig
+     *            - contains package of generated sources, as well as source and
+     *            destination folders
+     */
+    @Override
+    public void run(final Raml raml, final GeneratorConfig configuration) {
+        final File outputDirectory = createOutputDirectories(configuration);
```
Comment (purple52) [inline adapters/jms-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/JmsEndpointGenerator.java]: We already create and clear the output directory in the plugin - we should standardize on that approach, I think.

File: adapters/jms-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/JmsEndpointGenerator.java [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+    private String createJmsFilenameFrom(final String uri) {
+        return classNameOf(uri) + FILENAME_POSTFIX;
+    }
+
+    private String classNameOf(final String uri) {
+        return toCamelCase(uri);
+    }
+
+    private String toCamelCase(final String s) {
```
Comment (purple52) [inline adapters/jms-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/JmsEndpointGenerator.java]: Guava has a CaseFormat class for doing this sort of thing.

File: adapters/jms-adapter-generator/src/test/java/uk/gov/justice/raml/jms/core/JmsEndpointGeneratorTest.java [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+                                .with(action()
+                                        .with(ActionType.OPTIONS)
+                                        .withMediaType("application/vnd.structure.commands.test-cmd6+json"))
+                                .with(action()
+                                        .with(ActionType.PATCH)
+                                        .withMediaType("application/vnd.structure.commands.test-cmd7+json"))
+                                .with(action()
+                                        .with(ActionType.TRACE)
+                                        .withMediaType("application/vnd.structure.commands.test-cmd8+json")))
+                        .build(),
```
Comment (purple52) [inline adapters/jms-adapter-generator/src/test/java/uk/gov/justice/raml/jms/core/JmsEndpointGeneratorTest.java]: There is a static convenience action() method that can take an action type and a mediatype in one go, which would make this shorter.

File: adapters/pom.xml [hunk 7]
```diff
@@ -1,20 +1,19 @@
...
+
+    <artifactId>adapters</artifactId>
+    <packaging>pom</packaging>
+
+    <modules>
+        <module>rest-adapter-generator</module>
+        <module>jms-adapter-generator</module>
+        <module>adapters-test-utils</module>
+    </modules>
+</project>
```
Comment (purple52) [inline adapters/pom.xml]: Is this am EOL problem again? Only one line has been added...

File: adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/DefaultGenerator_ResourceCodeStructureTest.java [hunk 8]
```diff
@@ -67,6 +70,7 @@ public void shouldGenerateAnnotatedResourceInterface() throws Exception {
                 ).build(),
                 configurationWithBasePackage(BASE_PACKAGE));
 
+
         Class<?> interfaceClass = compiler.compiledInterfaceOf(BASE_PACKAGE);
 
         assertThat(interfaceClass.isInterface(), is(true));
```
Comment (purple52) [inline adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/DefaultGenerator_ResourceCodeStructureTest.java]: I think this is mainly a reordering of imports, but its difficult to see what has actually changed.

File: adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/DefaultGenerator_ResourceMethodBodyTest.java [hunk 9]
```diff
@@ -11,40 +40,13 @@
...
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
-import static org.raml.model.ActionType.POST;
-import static uk.gov.justice.services.adapters.test.utils.ActionBuilder.action;
-import static uk.gov.justice.services.adapters.test.utils.RamlBuilder.restRamlWithDefaults;
-import static uk.gov.justice.services.adapters.test.utils.ResourceBuilder.resource;
-import static uk.gov.justice.services.messaging.DefaultEnvelope.envelopeFrom;
-
 @RunWith(MockitoJUnitRunner.class)
 public class DefaultGenerator_ResourceMethodBodyTest {
```
Comment (purple52) [inline adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/DefaultGenerator_ResourceMethodBodyTest.java]: Same here - the imports are mainly the same, but the order has changed.

File: adapters/pom.xml (hunk 7 above)
Comment (jaceko) [inline adapters/pom.xml]: probably, not much I can do here..

File: adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/DefaultGenerator_ResourceMethodBodyTest.java (hunk 9 above)
Comment (jaceko) [inline adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/DefaultGenerator_ResourceMethodBodyTest.java]: probably a result of Eclipse's organise imports (Ctrl+Shift+O) to remove unused imports

File: adapters/jms-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/JmsEndpointGenerator.java (hunk 5 above)
Comment (jaceko) [inline adapters/jms-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/JmsEndpointGenerator.java]: yes we've considered it but it does not fit the the bill here (we need to camel case dot and/or slash delimited string)

File: adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/DefaultGenerator_ResourceCodeStructureTest.java (hunk 8 above)
Comment (jaceko) [inline adapters/rest-adapter-generator/src/test/java/uk/gov/justice/services/adapters/rest/generator/DefaultGenerator_ResourceCodeStructureTest.java]: this happened after I manually resolved a merge conflict

File: adapters/jms-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/JmsEndpointGenerator.java (hunk 4 above)
Comment (jaceko) [inline adapters/jms-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/JmsEndpointGenerator.java]: would a "create if empty" approach be acceptable here?

File: adapters/jms-adapter-generator/src/test/java/uk/gov/justice/raml/jms/core/JmsEndpointGeneratorTest.java (hunk 6 above)
Comment (jaceko) [inline adapters/jms-adapter-generator/src/test/java/uk/gov/justice/raml/jms/core/JmsEndpointGeneratorTest.java]: fixed

**Pull request closed without merging**